rust/datafusion/src/physical_plan/hash_aggregate.rs

```diff
@@ -288,6 +288,9 @@ fn group_aggregate_batch(
     // Make sure we can create the accumulators or otherwise return an error
     create_accumulators(aggr_expr).map_err(DataFusionError::into_arrow_external_error)?;
 
+    // Keys received in this batch
+    let mut batch_keys = vec![];
```

Comment:
Should this be a set rather than a vec since it is intended to track the unique set of keys in the batch?

Reply:
That's what I thought first, but this is checked already when `push`ing the keys to the vec.

Reply:
It checks for either an empty indices array (which means no rows yet with this key) or being the first row with this key in `or_insert_with`.